design/definitions.go

```diff
@@ -1413,6 +1423,20 @@ func (a *ActionDefinition) UserTypes() map[string]*UserTypeDefinition {
 	return types
 }
 
+// IterateHeaders iterates over the resource-level headers and then action-level headers,
+// calling the given iterator passing in each response sorted in alphabetical order.
+// Iteration stops if an iterator returns an error and in this case IterateHeaders returns that
+// error.
+func (a *ActionDefinition) IterateHeaders(it HeaderIterator) error {
+	// resource-level headers
+	err := iterateHeaders(a.Parent.Headers, it)
+	if err != nil {
+		return err
+	}
+	// action-level headers
+	return iterateHeaders(a.Headers, it)
+}
```

Comment:
Good point, I hadn't thought about overriding headers..

I had actually started out with merging them, but reverted because I ran into a problem with determining whether a header was `required`, since it could have come from two different places.. I'll have to give that more thought and try again.